pkg/certificates/letsencrypt.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright (C) 2021-2023, Kubefirst
+
+This program is licensed under MIT.
+See the LICENSE file for more details.
+*/
+package certificates
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/kubefirst/kubefirst-api/pkg/reports"
+)
+
+const letsDebugHost = "https://letsdebug.net/certwatch-query"
+
+// CheckCertificateUsage polls letsdebug to get information about used certificates
+func CheckCertificateUsage(domain string) error {
+	// Retrieve response from letsdebug regarding used certificates
+	req, err := http.NewRequest("GET", letsDebugHost, nil)
+	if err != nil {
+		return err
+	}
+	query := fmt.Sprintf(`WITH ci AS ( SELECT min(sub.CERTIFICATE_ID) ID, min(sub.ISSUER_CA_ID) ISSUER_CA_ID, sub.CERTIFICATE DER FROM (SELECT * FROM certificate_and_identities cai WHERE plainto_tsquery('%s') @@ identities(cai.CERTIFICATE) AND cai.NAME_VALUE ILIKE ('%%' || '%s' || '%%') LIMIT 10000 ) sub GROUP BY sub.CERTIFICATE ) SELECT ci.ID crtsh_id, ci.DER der FROM ci LEFT JOIN LATERAL ( SELECT min(ctle.ENTRY_TIMESTAMP) ENTRY_TIMESTAMP FROM ct_log_entry ctle WHERE ctle.CERTIFICATE_ID = ci.ID ) le ON TRUE, ca WHERE ci.ISSUER_CA_ID = ca.ID AND x509_notBefore(ci.DER) >= NOW() - INTERVAL '169 hours' AND ci.ISSUER_CA_ID IN (16418, 183267, 183283) ORDER BY le.ENTRY_TIMESTAMP DESC;`,
```

Comment:
this IDs harcoded is possible to move to another var to prevent forgot the reason why is important for other contributors

Reply:
Good feedback. thats a good todo